Tests/KituraCompressionTests/Decompression.swift

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Kitura
+import CZlib
+import LoggerAPI
+
+import Foundation
+
+enum UnzipError: Error {
+    case unzipError()
+}
+
+extension Data {
+    private func createStream() -> z_stream {
+        var stream = z_stream()
+        self.withUnsafeBytes { (bytes: UnsafePointer<Bytef>) in
+            stream.next_in = UnsafeMutablePointer<Bytef>(mutating: bytes)
+        }
+        stream.avail_in = uint(self.count)
+        
+        return stream
+    }
+    
+    public func decompress() throws -> Data {
+
+        let contiguousData = self.withUnsafeBytes { Data(bytes: $0, count: self.count) }
+        var stream = contiguousData.createStream()
+        var status = inflateInit2_(&stream, MAX_WBITS + 32, ZLIB_VERSION, Int32(DataSize.stream))
```

Comment:
The last parameter can just be `MemoryLayout<z_stream>.size`, then you can delete `struct DataSize` completely.